astropy/units/core.py

```diff
@@ -261,6 +269,59 @@ def add_enabled_equivalencies(self, equivalencies):
         equivalencies = _normalize_equivalencies(equivalencies)
         self._equivalencies |= set(equivalencies)
 
+    @property
+    def aliases(self):
+        return self._aliases
+
+    def set_enabled_aliases(self, aliases):
+        """
+        Set aliases for units.
+
+        Parameters
+        ----------
+        aliases : dict of str, Unit
+            The aliases to set. The keys must be the string aliases, and values
+            must be the `astropy.units.Unit` that the alias will be mapped to.
+
+        Raises
+        ------
+        ValueError
+            If the alias already defines a different unit.
+
+        """
+        self._reset_aliases()
+        self.add_enabled_aliases(aliases)
+
+    def add_enabled_aliases(self, aliases):
+        """
+        Add aliases for units.
+
+        Parameters
+        ----------
+        aliases : dict of str, Unit
+            The aliases to add. The keys must be the string aliases, and values
+            must be the `astropy.units.Unit` that the alias will be mapped to.
+
+        Raises
+        ------
+        ValueError
+            If the alias already defines a different unit.
+
+        """
+        for alias, unit in aliases.items():
+            if alias in self._registry and unit != self._registry[alias]:
+                raise ValueError(
+                    f"{alias} already means {self._registry[alias]}, so "
+                    f"cannot be used as an alias for {unit}.")
+            if alias in self._aliases and unit != self._aliases[alias]:
+                raise ValueError(
+                    f"{alias} already is an alias for {self._aliases[alias]}, so "
+                    f"cannot be used as an alias for {unit} too.")
```

Comment:
Removing the word "too" in the 2nd error message, I think these two if clauses can be "or"'d and have just one error to raise? I like the phrasing of the second better ("already is an alias for" vs. "already means")

Reply:
I had that initially but then the f-string becomes annoying, since one has to switch between `self._registry` and `self._aliases`...  I'll just delete "too"...